contracts/cw-core-factory/src/tests.rs

```diff
@@ -0,0 +1,113 @@
+    Box::new(contract)
+}
+
+fn cw_core_contract() -> Box<dyn Contract<Empty>> {
+    let contract = ContractWrapper::new(
+        cw_core::contract::execute,
+        cw_core::contract::instantiate,
+        cw_core::contract::query,
+    )
+    .with_reply(cw_core::contract::reply)
+    .with_migrate(cw_core::contract::migrate);
+    Box::new(contract)
+}
+
+#[test]
+pub fn test_set_admin() {
+    let mut app = App::default();
+    let code_id = app.store_code(factory_contract());
+    let cw20_code_id = app.store_code(cw20_contract());
+    let cw20_instantiate = cw20_base::msg::InstantiateMsg {
+        name: "DAO".to_string(),
+        symbol: "DAO".to_string(),
+        decimals: 6,
+        initial_balances: vec![],
+        mint: None,
+        marketing: None,
+    };
+
+    let instantiate = InstantiateMsg {};
+    let factory_addr = app
+        .instantiate_contract(
+            code_id,
+            Addr::unchecked("CREATOR"),
+            &instantiate,
+            &[],
+            "cw-core-factory",
+            None,
+        )
+        .unwrap();
+
+    // Instantiate core contract using factory.
+    let cw_core_code_id = app.store_code(cw_core_contract());
+    let instantiate_core = cw_core::msg::InstantiateMsg {
+        admin: None,
+        name: "DAO DAO".to_string(),
+        description: "A DAO that builds DAOs.".to_string(),
+        image_url: None,
+        automatically_add_cw20s: true,
+        automatically_add_cw721s: true,
+        voting_module_instantiate_info: ModuleInstantiateInfo {
+            code_id: cw20_code_id,
+            msg: to_binary(&cw20_instantiate).unwrap(),
+            admin: Admin::CoreContract {},
+            label: "voting module".to_string(),
+        },
+        proposal_modules_instantiate_info: vec![
+            ModuleInstantiateInfo {
+                code_id: cw20_code_id,
+                msg: to_binary(&cw20_instantiate).unwrap(),
+                admin: Admin::CoreContract {},
+                label: "prop module".to_string(),
+            },
+            ModuleInstantiateInfo {
+                code_id: cw20_code_id,
+                msg: to_binary(&cw20_instantiate).unwrap(),
+                admin: Admin::CoreContract {},
+                label: "prop module 2".to_string(),
+            },
+        ],
+        initial_items: None,
+    };
+
+    // multi-test does not support UpdateAdmin yet :(
+    // https://github.com/CosmWasm/cw-plus/blob/14f4e922fac9e2097a8efa99e5b71d04747e340a/packages/multi-test/src/wasm.rs#L477
+    let err = app
+        .execute_contract(
+            Addr::unchecked("CREATOR"),
+            factory_addr,
+            &ExecuteMsg::InstantiateCore {
+                msg: to_binary(&instantiate_core).unwrap(),
+                code_id: cw_core_code_id,
+            },
+            &[],
+        )
+        .unwrap_err();
+
+    assert!(matches!(
+        err.downcast().unwrap(),
+        cw_multi_test::error::Error::UnsupportedWasmMsg(_)
+    ))
```

Comment:
Until cw-plus supports this it might be nice to test this method on a non integration test way just using `mock_env` and then likes, calling the method directly, and then verifying that the returned response is what we expect. Here's an example of doing that from cw-plus: https://github.com/CosmWasm/cw-plus/blob/14f4e922fac9e2097a8efa99e5b71d04747e340a/contracts/cw20-base/src/contract.rs#L623-L638